anesthetic/plot.py

```diff
@@ -1263,17 +1340,13 @@ def hist_plot_2d(ax, data_x, data_y, *args, **kwargs):
     ymax = quantile(data_y, q[-1], weights)
     rge = kwargs.pop('range', ((xmin, xmax), (ymin, ymax)))
 
-    if levels is None:
-        pdf, x, y, image = ax.hist2d(data_x, data_y, weights=weights,
-                                     cmap=cmap, range=rge, vmin=vmin,
-                                     *args, **kwargs)
-    else:
-        bins = kwargs.pop('bins', 10)
-        density = kwargs.pop('density', False)
-        cmin = kwargs.pop('cmin', None)
-        cmax = kwargs.pop('cmax', None)
-        pdf, x, y = np.histogram2d(data_x, data_y, bins, rge,
-                                   density, weights)
+    bins = kwargs.pop('bins', 10)
+    density = kwargs.pop('density', False)
+    cmin = kwargs.pop('cmin', None)
+    cmax = kwargs.pop('cmax', None)
+    pdf, x, y = np.histogram2d(data_x, data_y, bins, rge,
+                               density, weights)
+    if levels is not None:
```

Comment:
Note, how I have dropped the use of `ax.hist2d`, here. 
Do we remember why we wanted this distinction? 
Tests are passing, so maybe this is an unnecessary complication?

Reply:
It's been there since the beginning in #46. I suspect it was due to wanting to use existing functionality where possible, but then having to implement something more specific for the level-based plotting.

Now that we're mirroring pandas more closely, the `ax.hist2d` is now accomplished (in weighted form!) by using `samples.plot.hist`, so we can indeed safely remove this. It's much neater now!